### DISCUSSION TITLE
SONOFF SNZB-01: strange action "bounce" messages

### DISCUSSION THREAD

**Discussion by vwegert** [Koenkk/zigbee2mqtt#24242] [Devices]
I'm in the process of moving my Zigbee installation towards a Zigbee2MQTT based setup. I'm using a number of SONOFF SNZB-01 pushbuttons, and whenever I press them, the following events are published to the MQTT topic:

```
(voltage, battery, linkquality removed)
[2024-10-06 18:07:34] info: 	z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/TestButton/action', payload 'single'
(last_seen, voltage, battery, linkquality removed)
[2024-10-06 18:07:34] info: 	z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/TestButton/action', payload ''
(last_seen removed)
[2024-10-06 18:07:34] info: 	z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/TestButton/action', payload 'single'
```
I'm fine with the first two messages - setting the action to "single" and then back again to nothing. I just don't understand why there is a final message setting the action back to "single" (and keeping it that way). 

This happens with multiple buttons. I've already tried to remove and re-add the devices, no change. Increasing the debounce value on the devices doesn't help either. Zigbee2mqtt version is 1.40.2 - does anyone have an idea how to get rid of these extraneous messages?

**Comment by vwegert** [comment]
Might be related to issue #24012

**Comment by vwegert** [comment]
Downgrading from 1.40.2 to 1.40.0 "solves" the issue

**Comment by Fafitude** [comment]
Yes, it's the same problem.

**Comment by Koenkk** [comment]
Could you provide the debug log on both 1.40.0 and 1.40.2 when triggering the switch?


See [this](https://www.zigbee2mqtt.io/guide/configuration/logging.html) on how to enable debug logging.
